rfcs/20190308-addons-proxy-maintainership.md

```diff
@@ -0,0 +1,174 @@
+granted after substantial contribution has been made in order to limit the 
+number of users with write permission. Contributions can come in the form of 
+issue closings, bug fixes, documentation, new code, or optimizing existing code. 
+Submodule maintainership can be granted with a lower barrier for entry as this will not 
+include write permissions to the repo.
+
+Additionally, Proxy maintainers are encouraged to co-maintain existing TensorFlow 
+Addons subpackages/submodules. If a potential maintainer chooses to do so, 
+there are a few things to consider first:
+
+* If the subpackage has an existing maintainer, the potential maintainer 
+must communicate with them first. The existing maintainer would need to 
+approve the co-maintainer role, and might give the potential maintainer 
+a specific task to complete before granting co-maintainership.
+
+* If the subpackage has major bugs, the potential maintainer 
+can provide fixes and expedite their inclusion as a listed
+proxy maintainer.
+
+### Privileges and Responsibitilies of Proxied Maintainers
+
+Subpackage maintainers will be granted write access to the repo and will be 
+designated in the [CODEOWNERS](https://help.github.com/en/articles/about-code-owners) 
+file.
+
+### Resolving Bugs
+
+Proxy maintainers are expected to handle bugs against the subpackages 
+they own. This includes resolving those issues once their fix is 
+merged by a proxy maintainer.
+
+If a proxy maintainer fails to provide sufficient support for their
+subpackage they will be designated as inactive. This status will be taken into 
+account during the periodic review of their subpackage/submodule.
+
+### Retiring from Proxy Maintainership
+
+If a proxy maintainer decides that they no longer wish to maintain one 
+or more of their subpackages, they **must** commit to the following procedure:
+
+* Send an email to addons@tensorflow.org and list all code that they are 
+no longer able to maintain. It is usually a good idea to shortly describe the 
+state of code, e.g. whether the packages have open bugs, whether they are difficult 
+to maintain, etc.
+
+* The proxy maintainer must remove themselves from the `CODEOWNERS.md` 
+file, with `<!--maintainer-needed-->` listed in the place of 
+maintainers. This will help other developers to find subpackages with no 
+maintainers.
+
+## Repository Growth and Review 
+### Periodic Review
+
+Given the nature of this repository, subpackages and submodules may become less 
+and less useful to the community as time goes on. In order to keep the 
+repository sustainable, we'll be performing bi-annual reviews of our code to 
+ensure everything still belongs within the repo. Contributing factors to this 
+review will be:
+
+1. Number of active maintainers
+2. Amount of OSS use
+3. Amount of issues or bugs attributed to the code
+
+Because we encourage TF-Addons to be used in production, we'll plan on a 
+graceful deprecation process, instead of a swift removal.
```

Comment:
We discussed the idea of a 2 phase deprecation process during the monthly meeting. We mark modules as "Going to be removed if nobody responds" or something similar. After half a year we do a "spring cleanup" and code that is not needed anymore will be removed.

Reply:
Agree. @qlzh727 described a 3 phase system after this post that I think gives us even more flexibility.